### PR TITLE
add tests for manifest rules

### PR DESCRIPTION
**Pull request by glommer**
We already had some, but not introduced more, rules about what paths
can be listed in the manifest.

Add tests for that. Also, it's overkill to use lit for this, when
rust unit tests would do.

### CODE REVIEW

**Comment by glommer** [comment]
FYI @dekimir

File: cli/src/project.rs [hunk 1]
```diff
@@ -302,3 +308,83 @@ pub(crate) fn project_exists(path: &Path) -> bool {
...
+endpoints = ["endpoints"]
+policies = ["policies"]
+"#,
+        );
+        let m = read_manifest_from(d.path()).unwrap();
+        m.models().unwrap();
+        m.policies().unwrap();
+        m.endpoints().unwrap();
+    }
+}
```
Comment (dekimir) [inline cli/src/project.rs:390]: Can we add an empty-string case?

File: cli/src/project.rs [hunk 2]
```diff
@@ -302,3 +308,83 @@ pub(crate) fn project_exists(path: &Path) -> bool {
...
+        m.policies().unwrap();
+        m.endpoints().unwrap();
+    }
+
+    #[should_panic(expected = "has to be a subdirectory")]
+    #[test]
+    fn parse_curr_dir_fails() {
+        let d = gen_manifest(
+            r#"
+models = ["./"]
```
Comment (dekimir) [inline cli/src/project.rs:364]: Is the slash ending significant? What about just `.`?

**Review by dekimir — APPROVED** [review]

File: cli/src/project.rs (hunk 1 above)
Comment (glommer) [reply cli/src/project.rs:390]: we can, but empty string is actually allowed (it is toml-equivalent to just not specifying it). It is legal not to specify either endpoints, models or policies

File: cli/src/project.rs (hunk 2 above)
Comment (glommer) [reply cli/src/project.rs:364]: they should both work the same. I didn't feel the need to add a test for that.

**Pull request merged**